src/mattext/models/classification.py

```diff
@@ -0,0 +1,188 @@
+        super().__init__(
+            cfg=cfg.model.representation,
+            special_tokens=cfg.model.special_tokens,
+            special_num_token=cfg.model.special_num_token,
+        )
+        self.fold = fold
+        self.local_rank = local_rank
+        self.representation = cfg.model.representation
+        self.data_repository = cfg.model.data_repository
+        self.cfg = cfg.model.finetune
+        self.context_length: int = self.cfg.context_length
+        self.callbacks = self.cfg.callbacks
+        self.tokenized_dataset = self._prepare_datasets(
+            self.cfg.path.finetune_traindata
+        )
+
+    def _prepare_datasets(self, subset: str) -> DatasetDict:
+        """
+        Prepare training and validation datasets.
+
+        Args:
+            train_df (pd.DataFrame): DataFrame containing training data.
+
+        Returns:
+            DatasetDict: Dictionary containing training and validation datasets.
+        """
+
+        def replace_none(example, replacement="[PAD]"):
+            for key, value in example.items():
+                if value is None:
+                    example[key] = replacement
+            return example
+
+        ds = load_dataset(self.data_repository, subset)
+        dataset = ds[self.fold].train_test_split(shuffle=True, test_size=0.2, seed=42)
+        dataset = dataset.filter(
+            lambda example: example[self.representation] is not None
+        )
+        return dataset.map(
+            partial(
+                self._tokenize_pad_and_truncate, context_length=self.context_length
+            ),
+            batched=True,
+        )
+
+    def _callbacks(self) -> List[TrainerCallback]:
+        """Returns a list of callbacks for early stopping, and custom logging."""
+        callbacks = []
+
+        if self.callbacks.early_stopping:
+            callbacks.append(
+                EarlyStoppingCallback(
+                    early_stopping_patience=self.callbacks.early_stopping_patience,
+                    early_stopping_threshold=self.callbacks.early_stopping_threshold,
+                )
+            )
+
+        if self.callbacks.custom_logger:
+            callbacks.append(CustomWandbCallback_FineTune())
+
+        callbacks.append(EvaluateFirstStepCallback)
+
+        return callbacks
+
+    def _compute_metrics(self, p: Any) -> Dict[str, float]:
+        preds = p.predictions[0] if isinstance(p.predictions, tuple) else p.predictions
+        preds_argmax = np.argmax(preds, axis=1)
+        labels = p.label_ids
+        precision, recall, f1, _ = precision_recall_fscore_support(labels, preds_argmax, average='weighted')
+        acc = accuracy_score(labels, preds_argmax)
+
+        # Compute ROC AUC
+        n_classes = preds.shape[1]
+        if n_classes == 2:
+            # Binary classification
+            roc_auc = roc_auc_score(labels, preds[:, 1])
+        else:
+            # Multi-class classification
```

Comment:
what is the multiclass classification you deal with? `is_metal` seems binary? Or is this just for "future proofing"?

Reply:
and you need to slice out one class because your model predicts the probability for True and False?

Reply:
I see, you have `num_labels` hardcoded  - in principle, one should be enough for binary

Reply:
for now hardcoding num_labels to two.
could be read from config later